src/services/audit.service.ts

```diff
@@ -0,0 +1,27 @@
+import { CryptoService } from 'jslib/abstractions/crypto.service';
+
+export class AuditService {
+
+    private url = 'https://api.pwnedpasswords.com/range/';
+
+    constructor(private cryptoService: CryptoService) {
+    }
+
+    async passwordLeaked(password: string) {
+        const hash = this.cryptoService.sha1(password);
+
+        const response = await fetch(this.url + hash.substr(0, 5));
+        const leakedHashes = await response.text();
+
+        const hashEnding = hash.substr(5);
```

Comment:
use this higher up so you dont have to substring twice

Reply:
The `substr` calls affect different portions of the hash. The first call returns the first 5 characters, while the other call returns the remaining portion of the hash.

Reply:
I see